seg_lapa/train.py

```diff
@@ -50,17 +57,37 @@ def test_step(self, batch, batch_idx):
         inputs, labels = batch
         outputs = self.model(inputs)
         loss = self.cross_entropy_loss(outputs, labels)
-
-        self.log("test_loss", loss, on_step=True, on_epoch=True, prog_bar=True, logger=True)
+        batch_loss = loss / len(batch[0])
+        wandb.log({"Test/Loss": batch_loss})
 
         return {
             "test_loss": loss,
         }
 
+    def training_epoch_end(self, outputs):
+        losses = torch.stack([x['loss'] for x in outputs])
+        gather_t = self.aggregate(losses)
+        gather_t = torch.flatten(torch.stack(gather_t))
+        loss = gather_t.mean()
+        wandb.log({"Train/Epoch Loss": loss})
+
+    def validation_epoch_end(self, outputs):
+        losses = torch.stack([x['val_loss'] for x in outputs])
+        gather_t = self.aggregate(losses)
+        gather_t = torch.flatten(torch.stack(gather_t))
+        loss = gather_t.mean()
```

Comment:
Same as above. We shouldn't need dist. functions in _step_end methods. I'll verify.